api/v1alpha4/cluster_types.go

```diff
@@ -60,6 +60,64 @@ type ClusterSpec struct {
 	// for provisioning infrastructure for a cluster in said provider.
 	// +optional
 	InfrastructureRef *corev1.ObjectReference `json:"infrastructureRef,omitempty"`
+
+	// This encapsulates the topology for the cluster.
+	// +optional
+	Topology *Topology `json:"topology,omitempty"`
```

Comment:
Can we add this later only when the field is functional behind a feature gate?

We should also call out that this needs the feature gate to be enabled otherwise it wouldn't work.

Reply:
I don't think that splitting API changes in two makes sense in practice (we need both of them ASAP to unblock further work), so I have added the feature flag in this PR.
Is that ok for you?